include/swift/PrintAsObjC/PrintAsObjC.h

```diff
@@ -21,12 +21,26 @@ namespace swift {
   class ModuleDecl;
   class ValueDecl;
 
+  enum class ObjcHeaderKind {
+    /// The header contains only public/open declarations from the module
+    Public,
+    /// The header contains only SPI declarations from the module. The header
+    /// includes the public header.
+    Private,
+    /// The header contains only internal declarations from the module. The header
+    /// includes both the public header and the private header.
+    Internal,
```

Comment:
Nitpick: the names public/internal/private suggest access control ordering (the one I just did). How about Public/Private/Project, to match Xcode?

(Also, traditionally "ObjC" rather than "Objc".)